Added new anti pattern blocking async code

dhedgepeth opened the pull request:
## 📋 Description

Adds new section documenting an anti-pattern for blocking async code, something we are seeing more frequently.

## ✅ Contributor Checklist

I've followed the [Umbraco Documentation Style Guide](https://docs.umbraco.com/contributing/documentation/style-guide) and can confirm that:

* [x] Code blocks are correctly formatted.
* [x] Sentences are short and clear (preferably under 25 words).
* [x] Passive voice and first-person language (“we”, “I”) are avoided.
* [x] Relevant pages are linked.
* [x] All links work and point to the correct resources.
* [x] Screenshots or diagrams are included if useful.
* [x] Any code examples or instructions have been tested.
* [x] Typos, broken links, and broken images are fixed.

## Product & Version (if relevant)

CMS 13-17

## 📚 Helpful Resources

* 🧾 [Umbraco Contribution Guidelines](https://docs.umbraco.com/contributing)
* ✍️ [Umbraco Documentation Style Guide](https://docs.umbraco.com/contributing/documentation/style-guide)

Commit:
Added block async calls in common pitfalls

sofietoft committed:
Minor formatting and grammar fixes

sofietoft committed:
Minor formatting and grammar fixes

sofietoft committed:
Minor formatting and grammar fixes

sofietoft committed:
Minor formatting and grammar fixes

sofietoft reviewed (changes requested):
Thanks for the PR @dhedgepeth ! 🙌 

Could you remove the changes to the v15 article from this PR? The version has been unpublished and removed, so that's why it's currently causing merge conflict.

Also, the vale linter is warning about the use of "relatively" and "significantly". Could you rewrite the sentences where these words are used? 

Finally: Do you need a technical review on this as well?

Commit:
Remove use of relatively and significantly

Commit:
Remove changes to v15 documentation

dhedgepeth commented:
Hey @sofietoft I've removed the changes to the v15 article and updated the articles to no longer use "significantly" or "relatively"😁

What's the right/best process for a technical review of the article? 
And thank you for taking a look, please let me know if you have any further suggestions! 🙌

sofietoft commented:
Great, @dhedgepeth ! Thanks for resolving the merge conflicts 💪 

Post a link to the PR in the umbraco-cms channel on our internal Slack.
Then someone from the CMS teams will help us with a review 🙌

kjac commented:
@sofietoft ah but this is not especially relevant to Cloud projects. It is relevant to any .NET project, and perhaps ASP.NET Core projects in particular, since they're likely serving a lot of requests as public facing websites.

Now, don't get me wrong... I don't mind this section being added. I simply wanted to note that we have historically shunned from replicating MS docs in our own to the extent possible, to avoid having to keep our docs in line with the MS docs.

sofietoft commented:
I should have been more clear: I merely meant that it sounds like Davis is adding this due to them seeing issue related to it on Cloud sites. 
Therefor my suggestion to add a note about it, or move it to the Cloud docs.

I strongly agree that we should continue to shun from replicated the Microsoft docs in ours. We can't maintain those docs easily.

@meyntony If we could describe it shortly in our docs, and then reference the microsoft docs article, that would be a great solution! 
As long as you feel it covers what you need our customers to be aware of in terms of the issue you're seeing on Cloud 💪

sofietoft commented:
@dhedgepeth - Any news here?
